### ISSUE TITLE
extending

### ISSUE THREAD

**Issue by joeblew99** [mbana/monaco-go#2]
i need to extend this.

It seems that in the GUI you standard to implement some other functionality but then stopped.
Could you add a TODO type thing that explains where you want this to head. Its likely aligned with what i want too for a golang IDE.

**Comment by mbana** [comment]
i can make most of the monaco-editor run in the browser with much hacks, i.e., webpack.

i've said this already in the main monaco-editor and vscode repos, the apis need to more aligned otherwise, like you (you've spotted), unneccessary shims need to be written.

that and doing WebSocket -> TCP, which is what most Language Server only support, can be solved by using websockify. i couldn't get it to run, see https://github.com/novnc/websockify/issues/268, and other comments i've made. if you can, awesome! there are Dockerfiles that i made that proxied the WebSocket to the go-langserver TCP connection. if you have another idea on doing an automatic proxy to a TCP conn via a WebSocket. do let me know.

## long story short - where i want it to head

the benefit is, any language that is shorted natively in monaco-editor will have feature-rich editor on the web. look at polymer-editor, as a good example.

spin a langserver live on a cloud env to serve a repo.

## misc

the hard part is figuring out what is going to change on the vscode front, else you'll just be writing endless boilerplate code to marshall to and fro vscode and monaco-editor.

anywho...

**Comment by joeblew99** [comment]
mhhh... i see your point and boiler plate = lots of code to maintain. Life is too short for that..
 
here is what i woudl do:
1. write the client in go. i use gopherjs and Qt golang. THat gets me web, desktop and mobile covered. So now we have pure golang client.
- there are some decent wrapper for Ace editor in gopherjs that work fine.

2. then use a pure golang language server. https://github.com/sourcegraph/go-langserver

3. Now we have shared types between the client and server for web

its how i pretty much approach all projects now because you get to stay in golang and share types across boundaries. Add some protobuf code gen and your pumping

Then to support other languages write drivers within the golang language server to talk to the other language parsers (python js, etc etc).
